### PR TITLE
Fix #13242 by cleaning up error messages about `#[view]` function parameters

### LINKED ISSUE #13242: [Bug][move-compiler-v2] redundant and misspelled signer parameter error
b# 🐛 Bug

`&mut signer` parameter to a view function yields 2 errors, leading to potential confusion.  The second one is also misspelled ("paramter") making us look pkind of bad.

These both are generated in `aptos-move/framework/src/extended_checks.rs` and apply to the test in `aptos-move/e2e-move-tests/src/tests/attributes.rs`:
```
error: type `&mut signer` is not supported as a parameter type
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpRYTHVQ/sources/m.move:4:17
  │
4 │             fun view(_:&mut signer,value: u64): u64 { value }
  │                 ^^^^

error: view function cannot use the & signer paremter
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpRYTHVQ/sources/m.move:4:17
  │
4 │             fun view(_:&mut signer,value: u64): u64 { value }
  │                 ^^^^

test tests::attributes::test_view_attribute_with_mut_ref_signer - should panic ... ok
```

# To reproduce:
```
UB=1 MOVE_COMPILER_V2=true cargo test -p e2e-move-tests -- test_view_attribute_with_mut_ref_signer
```

This also occurs without compiler-v2, so priority reduced.

### PR DESCRIPTION
**Pull request by brmataptos**
## Description

Clean up error messages about `#[view]` function parameters.

See #13242 for old error messages.

New test output looks like:
```
error: `#[view]` function must return values
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpTfNXod/sources/m.move:4:17
  │
4 │             fun view(_value: u64) { }
  │                 ^^^^

test tests::attributes::test_view_attribute_on_non_view - should panic ... ok
test tests::attributes::test_view_attribute ... ok
BUILDING Package
error: type `&mut signer` is not supported as a transaction parameter type
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpFo5Hs8/sources/m.move:4:22
  │
4 │             fun view(_:&mut signer,value: u64): u64 { value }
  │                      ^

test tests::aggregator_v2_events::test_events_with_snapshots_not_emitted_on_abort::_2_expects ... ok
test tests::attributes::test_view_attribute_with_mut_ref_signer - should panic ... ok
BUILDING Package
error: `#[view]` function cannot use the `&signer` parameter
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpv6fPE1/sources/m.move:4:22
  │
4 │             fun view(_:&signer,value: u64): u64 { value }
  │                      ^

test tests::attributes::test_view_attribute_with_ref_signer - should panic ... ok
BUILDING Package
error: `#[view]` function cannot use the `signer` parameter
  ┌─ /var/folders/hx/_s_6fyh529bgmvl6z4m8m9fm0000gn/T/.tmpwicG9I/sources/m.move:4:22
  │
4 │             fun view(_:signer,value: u64): u64 { value }
  │                      ^

test tests::attributes::test_view_attribute_with_signer - should panic ... ok

```

## Type of Change
- [ ] New feature
- [x] Bug fix
- [ ] Breaking change
- [ ] Performance improvement
- [ ] Refactoring
- [ ] Dependency update
- [ ] Documentation update
- [ ] Tests

## Which Components or Systems Does This Change Impact?
- [ ] Validator Node
- [ ] Full Node (API, Indexer, etc.)
- [x] Move/Aptos Virtual Machine
- [ ] Aptos Framework
- [ ] Aptos CLI/SDK
- [ ] Developer Infrastructure
- [ ] Other (specify)

## How Has This Been Tested?
CI tests + some manual testing.

## Key Areas to Review
Is there any potential impact on other cases not tested?

### CODE REVIEW

File: aptos-move/framework/src/extended_checks.rs [hunk 1]
```diff
@@ -214,7 +214,7 @@ impl<'a> ExtendedChecker<'a> {
                 self.env.error(
                     loc,
                     &format!(
-                        "type `{}` is not supported as a parameter type",
+                        "type `{}` is not supported as an transaction parameter type",
```
Comment (rahxephon89) [inline aptos-move/framework/src/extended_checks.rs:217]: typo: "an" -> "a"

File: aptos-move/framework/src/extended_checks.rs [hunk 2]
```diff
@@ -148,7 +148,7 @@ impl<'a> ExtendedChecker<'a> {
                 if !ok {
                     self.env.error(
                         &fun.get_id_loc(),
-                        "`init_module` function can only take signers as parameters",
+                        "`init_module` function can only take `signer` values and immutable references as parameters",
```
Comment (rahxephon89) [inline aptos-move/framework/src/extended_checks.rs:151]: should be `signer` or the reference to a `signer` according to the check above?

File: aptos-move/framework/src/extended_checks.rs [hunk 3]
```diff
@@ -560,35 +560,39 @@ impl<'a> ExtendedChecker<'a> {
...
+                                    param_loc,
+                                    "`#[view]` function cannot use the `signer` parameter",
                                 )
                             }
-                        }
+                        },
+                        Type::Reference(mutability, inner) => {
+                            if let Type::Primitive(inner) = inner.as_ref() {
+                                if inner == &PrimitiveType::Signer
+                                    && mutability == &ReferenceKind::Immutable
```
Comment (rahxephon89) [inline aptos-move/framework/src/extended_checks.rs:584]: Any context on the reason why we need to check mutability here?

File: aptos-move/framework/src/extended_checks.rs (hunk 2 above)
Comment (brmataptos) [reply aptos-move/framework/src/extended_checks.rs]: must be immutable reference, I figured out how to word it better.

File: aptos-move/framework/src/extended_checks.rs (hunk 3 above)
Comment (brmataptos) [reply aptos-move/framework/src/extended_checks.rs:590]: added a comment.

**Review by rahxephon89 — APPROVED** [review]

**Review by vineethk — APPROVED** [review]
LGTM.

**Pull request merged**